### ISSUE TITLE
Allow defining start/stop command  setting per connection

### ISSUE THREAD

**Issue by kwibus** [atcomputing/guacamole-trigger#3]
Now gucamole-trigger settings apply to all connections.
But you might want only use this plugin on some connections.
Or have different  command for different connections.

This is easy to do for default authentication via *user-mappings.xml*.
But have to find out you can make it so you can also set it via mysql/posgress etc authentication

**Comment by adb014** [comment]
Yes I have a problem like this as well. I have both RDP and SSH connections, and I use "ForceCommand docker start --attach --interactive <CONTAINER>" in the sshd_config file to start the container. This allows both Guacamole and normal SSH clients to use the containers. So I only need guacamole-trigger for the RDP connections. What I'm currently doing is using hostnames like "cnt1-ssh" and "cnt2-rdp" then ignoring the START_COMMAND (and STOP_COMMAND for what it matters) if the hostname ends in "-ssh".

The issue with this is that my containers aren't in the same docker network as the guacamole, for security reasons. Therefore using the hostname means that I also need a DNS as well in the same network as the guacamole to resolve the hostnames to IP. 

It would be so much easier if guacamole-trigger also exported the profile name so the START_COMMAND can see it rather than just hostname and guacamoleUsername.. I could then easily use the container name as the name of the profile and filter on that and leave the hostname with the correct IP addresses, and no longer need a local DNS

**Comment by kwibus** [comment]
Yes. I agree that would be nice. I am also looking for a way to specify `START_COMMAND`  options per connect.

It might be possible to pass all protocol option, and connection name to the script. hopefully this include made up protocol options. But have to lookup how that works with different authentication plugins. 

another thing to consider is the stop command. now it checks if multiple people/connections logged in into the same machine.
It only runs stop command if the number of connection to a host drop to 0. If you give different names to the same host.
it will see them as 2 different hosts, So it might stop the host even if there other connections still using it. 

Also if multiple connections start/stop the same host. but have different options. might be hard to debug.
So i have to think about it how i will do this.

# But about your use case.

If i understand you correctly:

* You want to start containers that you can connect to with both "ssh" and RDP. 
* You prefer to connect over IP

What is the problem with starting the container with both ssh and RDP connection?
this way it does not matter if you are logged in first with ssh or rdp. they will both start stop it based on use. 

Do you use tools like kubernetes, docker-compose docker swarm?
Can you share the config files  for your setup. so i might be able to reproduce the issue

**Comment by kwibus** [comment]
With latest version that i also mentions in #5.
It will now pass protocol and all other connection attributes.
those depend on protocol you choose. 
You can also  add dummy attributes to you connection config that would be passed to you start/stop scripts
But how to this depends on authentication provider used.
 
For your use case `protocol` would work to detect ssh.
